internal/internal_workflow_client.go

```diff
@@ -263,6 +263,89 @@ func (wc *workflowClient) SignalWorkflow(ctx context.Context, workflowID string,
 		}, serviceOperationRetryPolicy, isServiceTransientError)
 }
 
+// SignalWithStartWorkflow sends a signals to a workflow in execution, if workflow is not running or not found,
+// it starts workflow then sends the signal in transaction
+func (wc *workflowClient) SignalWithStartWorkflow(ctx context.Context, workflowID string, runID string, signalName string, signalArg interface{},
+	options StartWorkflowOptions, workflowFunc interface{}, workflowArgs ...interface{}) (*WorkflowExecution, error) {
+
+	signalInput, err := getEncodedArg(signalArg)
+	if err != nil {
+		return nil, err
+	}
+
+	if workflowID == "" {
+		workflowID = uuid.NewRandom().String()
+	}
+
+	if options.TaskList == "" {
+		return nil, errors.New("missing TaskList")
+	}
+
+	executionTimeout := common.Int32Ceil(options.ExecutionStartToCloseTimeout.Seconds())
+	if executionTimeout <= 0 {
+		return nil, errors.New("missing or invalid ExecutionStartToCloseTimeout")
+	}
+
+	decisionTaskTimeout := common.Int32Ceil(options.DecisionTaskStartToCloseTimeout.Seconds())
+	if decisionTaskTimeout < 0 {
+		return nil, errors.New("negative DecisionTaskStartToCloseTimeout provided")
+	}
+	if decisionTaskTimeout == 0 {
+		decisionTaskTimeout = defaultDecisionTaskTimeoutInSecs
+	}
+
+	// Validate type and its arguments.
```

Comment:
Validate type and its arguments. -> Validate the workflow type and its arguments.